docs/kb/semgrep-ci/jenkins-diff-scans.md

```diff
@@ -0,0 +1,185 @@
+            docker run \
+            -e SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN \
+            -v "$(pwd):$(pwd)" --workdir $(pwd) \
+            semgrep/semgrep semgrep ci '''
+      }
+    }
+  }
+}
+```
+
+This Jenkinsfile uses a `SEMGREP_APP_TOKEN` stored in the Jenkins instance credentials store. It does not set any other variables.
+
+### Set up a pipeline
+
+#### Under General
+
+1. Check the box "GitHub Project" and enter the URL for the project, `https://github.com/<namespace>/<project>/`
+2. Under Build Triggers, select the "GitHub Hook trigger for GITScm polling" box.
+
+#### Under Pipeline
+
+1. Definition: Pipeline script from SCM
+2. SCM: Git
+3. Repositories:
+    - Repository URL: `https://github.com/<namespace>/<project>/`
+    - Credentials: (select the appropriate credentials for your project)
+4. Branches to build: `refs/heads/main`
+5. Additional Behaviours: from the dropdown, select "Check out to specific local branch"
+    - Enter `**` for Branch name.
+6. Script Path: Jenkinsfile
+7. Check box: Lightweight checkout
+
+#### On GitHub
+
+If your Jenkins instance is configured to manage webhooks automatically on GitHub, these steps are not necessary. To review the settings and view the webhook URL, go to **Manage Jenkins > Configure System > GitHub**. Expand the <i class="fa-solid fa-circle-question"></i> next to **GitHub Server** to find the webhook URL, and review the configuration to see if hooks are managed automatically.
+
+1. Visit `https://github.com/<namespace>/<project>/settings/hooks`.
+2. Click **Add Webhook**.
+3. Under **Payload URL**, enter your Jenkins' instance webhook URL. Generally this is in the form `$JENKINS_BASE_URL/github-webhook/`.
+4. Select `application/json` for Content-Type.
+5. Select **Send me everything**.
+
+With configuration provided previously, findings in Semgrep AppSec Platform appear under the Jenkins project name, rather than under the typical GitHub name `<namespace>/<project>`. To change the name using `SEMGREP_REPO_NAME`, use this Jenkinsfile instead:
+
+```bash
+pipeline {
+  agent any
+  environment {
+    // Required for a Semgrep AppSec Platform-connected scan:
+    SEMGREP_APP_TOKEN = credentials('SEMGREP_APP_TOKEN')
+		// Set typical project (repo) name
+    SEMGREP_REPO_NAME = env.GIT_URL.replaceFirst(/^https:\/\/github.com\/(.*)$/, '$1')
+  }
+  stages {
+    stage('semgrep-scan') {
+      steps {
+        sh '''docker pull semgrep/semgrep && \
+            docker run \
+            -e SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN \
+            -e SEMGREP_REPO_NAME=$SEMGREP_REPO_NAME \
+            -v "$(pwd):$(pwd)" --workdir $(pwd) \
+            semgrep/semgrep semgrep ci '''
+      }
+    }
+  }
+}
+```
+
+## Adding diff-aware scans
+
+Semgrep diff-aware scans can be set up in several different ways using Jenkins. This example sets up a Multibranch Pipeline using `when` conditions in the Jenkinsfile, using a full scan configuration based on the example in the preceding section.
```

Comment:
I'd suggest replacing the first two paragraphs of this section with something like this:

> This example shows you how to set up full scans that run on the default branch and diff-aware scans that run on PR or MR branches. This is done with a Multibranch Pipeline using `when` conditions in the Jenkinsfile. The Multibranch pipeline provides access to useful variables for the configuration.

This is also minor, but instead of treating this section as additive to the first section where you set up a full scan, I think it might be less confusing if people just started here if they want both full and diff scans